### ISSUE TITLE
Create a type for managing RDS subnet groups

### ISSUE THREAD

**Issue by zabacad** [puppetlabs/puppetlabs-aws#373]
There is no type for managing DB subnet groups. These are sets of VPC subnets and are unique to RDS.

DB subnet groups consist of the following. The Puppet type can follow this closely. All parameters are required unless noted.
1. Name.
2. Description. (Required for some reason.)
3. Region
4. The VPC to create the DB subnet group in.
5. List of subnets. AWS requires there must be subnets for at least two different availability zones. These subnets must be part of the VPC.
6. [Optional] Tags.

Here is an example resource:

``` puppet
rds_subnet_group { 'production':
  ensure      => present,
  description => "RDS subnet group for production.",
  region      => 'us-east-1',
  vpc         => 'main',
  subnets     => [
    'rds-production-us-east-1a',
    'subnet-deadbeef',
  ],
  tags        => {
    environment => 'production',
  },
}
```

**Comment by DoctahPopp871** [comment]
@zabacad Have you started development on this resource ? If not I was going to start, as I have found the need for this. 
This change would certainly get the rds_instance resource closer to being able to spin up an RDS endpoint without any manual intervention.

**Comment by zabacad** [comment]
@DoctahPopp871 No, I haven't started. Go right ahead.

**Comment by DoctahPopp871** [comment]
@zabacad Cool Thanks! On It .

**Comment by DoctahPopp871** [comment]
I have a working version of this resource in my forked repository. Tested and working. Ironing out automated tests, and improving some of the code now. Should see a PR here soon!

**Comment by zleswomp** [comment]
Hows this going @DoctahPopp871?  I'd be happy to help with tests if that's all that remains.  This came up recently for us, and I'd love to bring this in.

**Comment by DoctahPopp871** [comment]
Howdy @zleswomp, I switched gigs and kind of drifted away from this project. Heres a PR with a functioning type and provider. Im not sure if Ill have time to wrap this up, so feel free to run with it.  https://github.com/puppetlabs/puppetlabs-aws/pull/422

**Comment by zleswomp** [comment]
Great @DoctahPopp871, thanks for rescuing the code!  I'll have a review on #422 today.

**Comment by zleswomp** [comment]
I believe this is resolved with the merge of #422.

**Comment by DoctahPopp871** [comment]
@zleswomp awesome.
